logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogStreamWriterImpl.java

```diff
@@ -39,7 +39,7 @@ public final class LogStreamWriterImpl implements LogStreamRecordWriter {
   private BufferWriter metadataWriter;
   private BufferWriter valueWriter;
 
-  LogStreamWriterImpl(final int partitionId, final Dispatcher logWriteBuffer) {
+  public LogStreamWriterImpl(final int partitionId, final Dispatcher logWriteBuffer) {
```

Comment:
:x: I understand we use it in the broker tests now, but this should be implementation detail for the broker - is there no way to just get a `LogStreamWriter` for those tests? I would ideally not want to depend on `impl` stuff from outside the module.

Reply:
I think the test which uses this should completely replaced by something else. We totally depend on internals and add more glue code for logstorage as a separate rule. I feel this can totally diverge since we rely on test code and not on real production code.

Reply:
Please either do it in this scope or create a follow up issue for this :+1:

Reply:
I created https://github.com/camunda/zeebe/issues/9244 for this